crates/store/src/db/mod.rs

```diff
@@ -40,6 +41,17 @@ pub struct Db {
     pool: deadpool_diesel::Pool<ConnectionManager, deadpool::managed::Object<ConnectionManager>>,
 }
 
+/// Describes the value of an asset for an account ID at `block_num` specifically.
+///
+/// If `asset` is `None`, the asset was removed.
+#[derive(Debug, Clone)]
+pub struct AccountVaultValue {
+    pub block_num: BlockNumber,
```

Comment:
Same question as in one of the other comments: why do we need `block_num` here but we didn't need it for `StorageMapValue` in https://github.com/0xMiden/miden-node/pull/1140?

To me it feels right to include the `block_num` (because it makes the list of updates a less ambiguous), but then we also should make the same changes in https://github.com/0xMiden/miden-node/pull/1140.

Also, this struct should probably be defined next to `StorageMapValue` (which is defined in `src/db/models/queries/accounts.rs` - though, I'm not sure if that's the right place for it).